src/VirtoCommerce.Platform.Data/Settings/SettingsManager.cs

```diff
@@ -170,15 +172,16 @@ public virtual async Task SaveObjectSettingsAsync(IEnumerable<ObjectSettingEntry
                     .Where(x => settingNames.Contains(x.Name))
                     .ToListAsync());
 
-                foreach (var setting in objectSettings.Where(x => x.ItHasValues))
+                var validator = new ObjectSettingEntryValidator();
+                foreach (var setting in objectSettings.Where(x => x.ItHasValues).Where(x => validator.Validate(x).IsValid))
```

Comment:
Here I think we should not save valid entities and skip not valid, we should throw an exception in case any value is invalid.
Otherwise, data could be saved in an unpredicted manner from the user POV.